### DISCUSSION TITLE
`extractVersion` Guava with release version suffix

### DISCUSSION THREAD

**Discussion by rickie** [renovatebot/renovate#22654] [Request Help]
### How are you running Renovate?

Self-hosted 

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.111.0

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

It didn't work.

### Wanted end result.

[These changes](https://github.com/rickie/renovate-guava-extractversion/pull/2/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8L16-R16) should be like this:
```xml
-     <version>30.0-jre</version>
+     <version>32.0.1-jre</version>
```

Additionally, we would like to make sure that Renovate picks up on the changelogs of Guava and adds that to the commit as well.

### What you tried so far.

Tried different ways of extracting the version to see if there is a way in which Renovate can re-use that suffix. We couldn't find a nice solution for this yet...

There is a reproduction repository that contains a minimal reproducible example: https://github.com/rickie/renovate-guava-extractversion.

### Relevant debug logs

<details><summary>Logs</summary>

```
 {
           "branchName": "renovate/guava-32.x",
           "prNo": 2,
           "prTitle": "Upgrade Guava 30.0-jre -> 32.0.1",
           "result": "pr-created",
           "upgrades": [
             {
               "datasource": "maven",
               "depName": "Guava",
               "displayPending": "",
               "fixedVersion": "30.0-jre",
               "currentVersion": "30.0-jre",
               "currentValue": "30.0-jre",
               "newValue": "32.0.1",
               "newVersion": "32.0.1",
               "packageFile": "pom.xml",
               "updateType": "major",
               "packageName": "com.google.guava:guava"
             }
           ]

```

</details>

**Comment by rarkins** [comment]
If the precision was consistent (e.g. always 2 places or always 3 places) then you could set versioning=docker and it would already work. Instead, you will want to use `regex` versioning where the patch part is optional. Something like this:

```
  "versioning": "regex:^(?<major>\\d+)\\.(?<minor>\\d+)(\\.(?<patch>\\d+))?-(?<compatibility>.*)$"
```

i.e. match major.minor and then optionally .patch, then -compatibility

**Reply by rickie** [reply]
Thanks for answering so quickly.

Yes this works for the prefix. However, now the changelogs are not added to the commit, and we would like to have that as well.
I tried several options for the `extractVersion` but when I configure this, the PRs get closed....

Can you elaborate on how I can fix this?

**Reply by rarkins** [reply]
I'm not sure there's any easy way to fix it. There's certainly not a simple answer I can give you. You'd need to run Renovate locally (from source) so that you can debug the changelog retrieval process to see where the problem occurs, and only then can you have an idea of if it's reasonably fixable

**Reply by viceice** [reply]
there is an open issue about providing an extractVersion option just for changelogs, so renovate can map `v1.2.3-slim` to a `1.2.3` release
